luatest/server.lua

```diff
@@ -742,8 +742,7 @@ end
 
 local function exec_tail(ok, ...)
     if not ok then
-        local _ok, res = pcall(json.decode, tostring(...))
-        error(_ok and res or ..., 0)
```

Comment:
This case isn't broken. Note that the function passed to `server:exec()` is called with `pcall`. Before this patch, we re-raised the error so we had to serialize it in JSON to be marshaled correctly through `net.box.eval`. This behavior was introduced in commit e6a2093780380f3e25c906752af466f3c9922272. This patch effectively reverts that commit. Now, any error raised by the function is returned from `net.box.eval`, not re-raised. Hence no manual serialization is required.

Commit e6a2093780380f3e25c906752af466f3c9922272 replaced `return` with `error` in the error case to prevent Tarantool from overwriting the error if there's an active transaction. However, the fix could be much simpler - it could just rollback the active transaction with `box.rollback()` before return. That's what this patch does.

Reply:
I've missed this `error` -> `return` movement in this patch. Thanks for the clarification!